proof_test.go

```diff
@@ -51,6 +51,1029 @@ var proofTestDefs = []struct {
+	{key: "fc25f30a05a768f4ca1dcafe9ddd7d0d108d001121238080705f4535", value: "c296e1de6fea3c1dfe63885f42518ac6b33ecc1ae12deac2b0f9a9df0cd10394"},
+	{key: "0a820b48ebfa30505b5ebec6fd41e81562ac3e73b54b994a6f7c07eb", value: "53b4126fce21de1925e431c937ca4bdb742508f26e7771324176013a1278b7e9"},
+	{key: "28c2c5e870badc9c12ca2e4519838772f7246d9b75d995cf626a09f4", value: "ed162f74ede886bb34896b1099c55070815818340773d270e9e8171f0711b711"},
+	{key: "0a820b48ebfa30505b5ebec6fd41e81562ac3e73b54b994a6f7c07eb", value: "0c878d5f965460ad491a935e865a417169161bc0fef03b1ad8da2cc17d9233e7"},
+	{key: "a277800d50a7a94afa7ab0c6db258d96debad650a507443b1e1bcad9", value: "64ff8aa6adc7e92bb70739051207b76268ce44334f9f925c5ec82f71a0c98aba"},
+	{key: "d405c3d46745ec1ea167342b8d014c53e4929c0dac518951a18a906c", value: "b6f2b9a9094b21ef5eb5132de895e644036219ff685d6159150d0cd01d97993b"},
+	{key: "ec5d793c475716d2381f3f6f0fd0e1559276e6e236238e699f4f823f", value: "cfb692597cfe249e4b455257f932b9d82d9ab13f1ad97229e916cf5016839148"},
+	{key: "d7a989e95ba4206a95418a72f9501863e84f2a639dc0f1085189b425", value: "7705add7f28227d1f5c3a5e076c38ad06eb9d3d70dd2cb5c40ed2d49e383c952"},
+	{key: "5a442b914cd286ae99b80fb4bf7135ff993beae34e6e0748d82fd779", value: "ce0261ef0bd2c547b2ae5ed1f9d5f0509a760258f2a7fa15e236c366d24ead3a"},
+	{key: "b5dd9ebf3980e60770579cfa718260003ded58af46266640aca0793b", value: "589e8b63d6f380beaadca0179d9b37e6ceb404c184c82ce850b89346c7ff32e8"},
+	{key: "98a77863cdd6408ca568ce5a6a5b371817a018f258a4b35e2e211176", value: "b13f68b9b710aebb648c8f470241dd204dc1a12cf95e3cbab8440de15a481b81"},
+	{key: "9a9f9ad845523a55422b5720db7abbdf827733b0945a3aafbd5e33f9", value: "65f41499f8328306d8deeb300080c48b5454adc2886f02fe20aee76244371082"},
+	{key: "3cb106a8308c9af70ea7a1d158f730a10b2ad6b69fee1a223ee0e73d", value: "e64fe15ce1425f5d582c71ddf5e05bfde37c84dcaefb6347d579ac52760e2c15"},
+	{key: "e85c63dda97ca4b78b5141f08c0a6f09a31528e3eddff55eafd9a7db", value: "6aef3d08cd76174714b425650270faa775de30f377736554140e1b8fa679f30a"},
+	{key: "7e9eb3cb3c8cbeb8ca4fc02cfeab30cafba55d005d7b92f18b3d5cbd", value: "5284cc6c3bafb94a8cc1a4a225bbf89964560fb798f0e2b690f2d73f51ba4cd8"},
+	{key: "02b0b9f8cdc320c2f2aa033ca339669e64774560fa067ede1f9418fa", value: "63a37fc38d2cf0ebb4445560f4ccd466a4c5295ca24f7bf3d3f7a7393aa61924"},
+	{key: "44b61a843e221b2755817ebbcad33b3b35ee166757e13603300aa1f0", value: "58388a64f20363ee7b994e1cc9f2182148533be5c5c432bfa359e9b0978ea96e"},
+	{key: "26a7f5b5b96e06f4c74ad18609aebacb1669a2d3df54b0a297d337b5", value: "818336f3a29e0e019ba5cd8e7720dc3a2e90a500cbe0233d4e92dcdff7be9bdc"},
+	{key: "a9621001478c486c1f23fc043e2faaa80a617891186108e34e36adf5", value: "4111429c2f4a1ae2cb570e424818de75a58fba4562894158b291de29e9a0d6c6"},
+	{key: "57123012015aeb86ca415a14c3d6dbb2614a238c1e49adedbe9afe1b", value: "39e207c598c3c23800c0aa5545402fbf7f019a2be3ca4fbf3994c071a10b4eb4"},
+}
```

Comment:
Minor nitpick, but please move these closer to (or even inside) the function that uses them